### PR TITLE
Fix #2042 - Add scroll-snap like behaviour on landing page

### LINKED ISSUE #2042: [U.I 2.0][Landing Page] Scroll-snap on lading page.
**What would you like to be added**:
In #1976 we are updating our lading page.
@huynguyez gave us https://www.tesla.com/ as a reference, and the site seems to be using some Scroll-Snap.

@DukeManh made some tests and found some issues with the implementation: 

From @DukeManh (in #1921)
>after playing around with scroll snap, I don't think scroll snap would work well for our page layout. You can see what I mean here https://codepen.io/Ducmanhbui/pen/oNYreYx. If you can come up with a workaround, that'll be great; otherwise, we might have to close this issue.

Sites with some references about scroll-snap:

https://css-tricks.com/practical-css-scroll-snapping/
https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-padding-bottom
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap

@humphd, we would appreciate some help/guidance on how to implement the scroll-snap.

### PR DESCRIPTION
**Pull request by DukeManh**
## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Fixes #2042

   - This solution is limited. It only works if you scroll and release your touchpad or mouse wheel. It won't work if you keep scrolling (overriding the scroll trigger). I will try to improve this later. 
   - The page will scroll down to the timeline on passing `100px` threshold, and scroll up on passing `height of banner - 50px`  thresold. 

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by humphd** [comment]
```

./src/components/Banner.tsx:1:26
--
16:46:30.265 | Type error: Cannot find module 'smoothscroll-polyfill' or its corresponding type declarations.
16:46:30.266 | > 1 \| import smoothscroll from 'smoothscroll-polyfill';
16:46:30.266 | \|                          ^
16:46:30.266 | 2 \| import { useEffect, useState, useRef } from 'react';
16:46:30.266 | 3 \| import {
16:46:30.266 | 4 \|   makeStyles,

```

**Comment by DukeManh** [comment]
> ```
> 
> ./src/components/Banner.tsx:1:26
> --
> 16:46:30.265 | Type error: Cannot find module 'smoothscroll-polyfill' or its corresponding type declarations.
> 16:46:30.266 | > 1 \| import smoothscroll from 'smoothscroll-polyfill';
> 16:46:30.266 | \|                          ^
> 16:46:30.266 | 2 \| import { useEffect, useState, useRef } from 'react';
> 16:46:30.266 | 3 \| import {
> 16:46:30.266 | 4 \|   makeStyles,
> ```
`
@humphd. Npm run build doesn't fail locally. I don't know why `@types/smoothscroll-polyfill` is not working.

**Comment by humphd** [comment]
You must have something installed locally that you didn't add to package.json, and is missing on Vercel

File: package.json [hunk 1]
```diff
@@ -90,6 +91,7 @@
     "pino-pretty": "4.7.1",
     "sanitize-html": "2.3.2",
     "set-interval-async": "1.0.34",
+    "smoothscroll-polyfill": "0.4.4",
```
Comment (PedroFonsecaDEV) [inline package.json:94]: You need to install [here](https://github.com/Seneca-CDOT/telescope/blob/master/src/web/package.json)
Not in the root `package.json`

Comment (DukeManh) [reply package.json]: Right, I just realized

**Review by birtony — APPROVED** [review]
I like how this works

**Review by yuanLeeMidori — APPROVED** [review]
Work well locally!

**Comment by humphd** [comment]
We need a rebase on this.

**Comment by DukeManh** [comment]
This solution is limited. It only works if you scroll and release your touchpad or mouse wheel. It won't work if you keep scrolling (overriding the scroll trigger). I will try to improve this later.

**Comment by DukeManh** [comment]
I just rebased, can I get another review?

**Review by PedroFonsecaDEV — DISMISSED** [review]

**Comment by humphd** [comment]
It looks like this is still behind (needs another merge conflict fixed).  If you do it, I'll review and land with you.

**Comment by DukeManh** [comment]
Can you help merge when it's ready?

**Pull request merged**